### PR TITLE
Fix split and array_split with indices overrun

### LINKED ISSUE #2163: CUDARuntimeError printing empty split
`CUDARuntimeError` printing empty split with out-of-bounds indices.

Code to reproduce follows.
```python
import cupy as cp
a = cp.array([])
out = cp.split(a, [1])  # Indices must be 1, i.e. out-of-bounds to reproduce the error.

# Trying to print `out` yields an error while numPy would print `[array([], dtype=float64), array([], dtype=float64)]`
print(out) 
```

```
Traceback (most recent call last):
  File "test.py", line 4, in <module>
    print(out)
  File "cupy/core/core.pyx", line 1290, in cupy.core.core.ndarray.__repr__
  File "cupy/core/core.pyx", line 1388, in cupy.core.core.ndarray.get
  File "cupy/cuda/memory.pyx", line 422, in cupy.cuda.memory.MemoryPointer.copy_to_host
  File "cupy/cuda/runtime.pyx", line 265, in cupy.cuda.runtime.memcpy
  File "cupy/cuda/runtime.pyx", line 145, in cupy.cuda.runtime.check_status
cupy.cuda.runtime.CUDARuntimeError: cudaErrorInvalidValue: invalid argument
```

### PR DESCRIPTION
**Pull request by takagi**
Close #2163.

`cupy.split` and `cupy.array_split` raise some error when they take indices that have an index exceeds the target axis' dimension. This PR fix it to make them return empty arrays for such sections as NumPy does.

### CODE REVIEW

**Comment by emcastillo** [comment]
Jenkins, test this please

File: cupy/core/_routines_manipulation.pyx [hunk 1]
```diff
@@ -435,6 +435,7 @@ cpdef array_split(ndarray ary, indices_or_sections, Py_ssize_t axis):
     if ary.size == 0:
         stride = 0
     for index in indices:
+        index = min(index, size)
```
Comment (emcastillo) [inline cupy/core/_routines_manipulation.pyx]: using built-in min could synchronize the device here

**Comment by emcastillo** [comment]
Please fix C

**Comment by takagi** [comment]
Okay, and I got what you thought!

**Comment by takagi** [comment]
I've fixed to require NumPy>=1.11 for the failing tests.
https://docs.scipy.org/doc/numpy-1.11.0/release.html#futurewarning-to-changed-behavior

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**